### ISSUE TITLE
404 Resource Not Found for every request

### ISSUE THREAD

**Issue by nikriek** [benedikt/layer-ruby#12]
Hi there, each request returns a 404 error. A request interception revealed that the app id isn't filled in: `POST https://api.layer.com/apps//webhooks`. The client itself seems to be properly configured, as I can access the app id and platform token. Any ideas?

**Comment by benedikt** [comment]
Hi @nikriek!

That definitely sounds like a weird issue. By default, all resources use `Layer::Resource.client` which defaults to `Layer::Client::Platform.new`. Does the default client have the app_id and platform token set correctly? 

Cheers,

Benedikt

**Comment by nikriek** [comment]
The client is configured in my Rails app using secrets.yml. I'm using `Layer::Client` in rake task. This code here works:

```
puts Layer::Client.app_id # => "layer:///apps/…"
puts Layer::Client.token # => "<TOKEN>"
```

**Comment by benedikt** [comment]
It seems that the `Layer::Resource.client` is initialized before `app_id` and `token` are set. Are you using an initializer to set the values?

**Comment by nikriek** [comment]
Yes, I have a file `initalizers/clients.rb`:
```
Layer::Client.configure do |config|
  config.app_id = Rails.application.secrets.layer_app_id
  config.token  = Rails.application.secrets.layer_token
end
```

**Comment by benedikt** [comment]
As a band-aid you can add a line like this to your initializer:

`Layer::Resource.client = Layer::Client::Platform.new`

Other than that, I think there must be something in your code that uses `Layer::Resource.client` before it is configured. Sorry I can't be of more help with this issue right now.

**Comment by nikriek** [comment]
I just upgraded to the newest version to use webhooks. It had worked before with Conversations etc… strange

**Comment by benedikt** [comment]
I was able to reproduce the issue locally. I'm looking into how to fix it.

**Comment by nikriek** [comment]
Thanks a lot, also for maintaining this library! :+1:

**Comment by benedikt** [comment]
I just pushed a possible fix. Can you please verify it works for you by using the current version in master? 

Just add `gem 'layer-ruby', github: 'benedikt/layer-ruby'` to your `Gemfile` and run `bundle update layer-ruby`

**Comment by nikriek** [comment]
It's working 🎉

**Comment by benedikt** [comment]
Great! I just published v0.5.2 :)
